File: week08/우현서/1920_우현서.py
import sys

n = int(sys.stdin.readline())
numbers = set(map(int, sys.stdin.readline().split()))
m = int(sys.stdin.readline())
find = list(map(int, sys.stdin.readline().split()))

for x in find:
    if x in numbers:
        print(1)
    else:
        print(0)

# 리스트로 여부 확인 안됨 -> 입력에 음수 있기 때문

# 이분 탐색  사용
# n = int(sys.stdin.readline())
# numbers = list(map(int, sys.stdin.readline().split()))
# m = int(sys.stdin.readline())
# find = list(map(int, sys.stdin.readline().split()))
#
# numbers.sort()
#
# for x in find:
#     left, right = 0, n-1
#     check = False
#     while left <= right:
#         mid = (left + right) // 2
#         if numbers[mid] == x:
#             check = True
#             break
#         elif numbers[mid] < x:
#             left = mid + 1
#         else:
#             right = mid - 1
#
#     if check:
#         print(1)
#     else:
#         print(0)

File: week38/우현서/1920_우현서.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.StringTokenizer;

// n개 정수 배열 a에 x 존재여부 출력

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        StringTokenizer st = new StringTokenizer(br.readLine());
        int[] A = new int[n];
        for (int i = 0; i < n; i++) {
            A[i] = Integer.parseInt(st.nextToken());
        }
        Arrays.sort(A);
//        System.out.println(Arrays.toString(A));

        int m = Integer.parseInt(br.readLine());
        st = new StringTokenizer(br.readLine());
        for (int i = 0; i < m; i++) {
            int value = Integer.parseInt(st.nextToken());
            System.out.println(binarySearch(A, value));
        }
    }

    public static int binarySearch(int[] A, int target) {
        int left = 0;
        int right = A.length - 1;
        while (left <= right) {
            int mid = (left + right) / 2;
            if (A[mid] == target) {
                return 1;
            } else if (A[mid] < target) {
                left = mid + 1;
            } else {
                right = mid - 1;
            }
        }
        return 0;
    }
}